internal/literals/obfuscators.go

```diff
@@ -79,6 +121,141 @@ func operatorToReversedBinaryExpr(t token.Token, x, y ast.Expr) *ast.BinaryExpr
+	count := minCount + obfRand.Intn(maxCount-minCount)
+
+	keys := make([]*extKey, count)
+	for i := 0; i < count; i++ {
+		keys[i] = randExternalKey(obfRand)
+	}
+
+	return keys
+}
+
+func addTrashExternalKeys(obfRand *mathrand.Rand, keys []*extKey) []*extKey {
+	if len(keys) >= maxExternalKeyCount {
+		return keys
+	}
+
+	trashCount := maxExternalKeyCount - len(keys)
+	if trashCount > 1 {
+		trashCount = 1 + obfRand.Intn(trashCount)
+	}
+	for i := 0; i < trashCount; i++ {
+		keys = append(keys, randExternalKey(obfRand))
+	}
+
+	obfRand.Shuffle(len(keys), func(i, j int) {
+		keys[i], keys[j] = keys[j], keys[i]
+	})
+
+	return keys
+}
+
+func externalKeysToParams(obfRand *mathrand.Rand, keys []*extKey) (params *ast.FieldList, args []ast.Expr) {
+	params = &ast.FieldList{}
+	for _, key := range addTrashExternalKeys(obfRand, keys) {
+		name := key.Name()
+		if !key.Used() {
+			name.Name = "_"
+		}
+		params.List = append(params.List, &ast.Field{
+			Names: []*ast.Ident{name},
+			Type:  key.Type(),
+		})
+		args = append(args, &ast.BasicLit{
+			Kind:  token.INT,
+			Value: fmt.Sprint(key.value),
+		})
+	}
+	return
+}
+
+func dataToByteSliceWithExternalKeys(obfRand *mathrand.Rand, data []byte, externalKeys []*extKey, rarity extKeyRarity) ast.Expr {
+	if !rarity.Try(obfRand) {
+		return ah.DataToByteSlice(data)
+	}
+
+	count := len(data) / 2
+	if count > maxExternalKeyCount*2 {
+		count = maxExternalKeyCount * 2
+	}
+
+	var stmts []ast.Stmt
+	for i := 0; i < count; i++ {
+		key := externalKeys[obfRand.Intn(len(externalKeys))]
+		key.AddRef()
+		idx, op, b := obfRand.Intn(len(data)), randOperator(obfRand), obfRand.Intn(key.bits/8)
+		data[idx] = evalOperator(op, data[idx], byte(key.value>>(b*8)))
+		stmts = append(stmts, &ast.AssignStmt{
+			Lhs: []ast.Expr{ah.IndexExpr("data", ah.IntLit(idx))},
+			Tok: token.ASSIGN,
+			Rhs: []ast.Expr{
+				operatorToReversedBinaryExpr(op,
+					ah.IndexExpr("data", ah.IntLit(idx)),
+					ah.CallExprByName("byte", &ast.BinaryExpr{
+						X:  key.Name(),
+						Op: token.SHR,
+						Y:  ah.IntLit(b * 8),
+					}),
+				),
+			},
+		})
+	}
+
+	slices.Reverse(stmts)
+
+	stmts = append([]ast.Stmt{ah.AssignDefineStmt(ast.NewIdent("data"), ah.DataToByteSlice(data))}, append(stmts, ah.ReturnStmt(ast.NewIdent("data")))...)
+	return ah.LambdaCall(&ast.ArrayType{Elt: ast.NewIdent("byte")}, ah.BlockStmt(stmts...))
+}
+
+func byteLitWithExternalKeys(obfRand *mathrand.Rand, val byte, externalKeys []*extKey, rarity extKeyRarity) ast.Expr {
+	if !rarity.Try(obfRand) {
+		return ah.IntLit(int(val))
+	}
+
+	key := externalKeys[obfRand.Intn(len(externalKeyRanges))]
+	key.AddRef()
+
+	op, b := randOperator(obfRand), obfRand.Intn(key.bits/8)
+	newVal := evalOperator(op, val, byte(key.value>>(b*8)))
+	return operatorToReversedBinaryExpr(op,
+		ah.CallExprByName("byte", ah.IntLit(int(newVal))),
+		ah.CallExprByName("byte", &ast.BinaryExpr{
+			X:  key.Name(),
+			Op: token.SHR,
+			Y:  ah.IntLit(b * 8),
+		}),
+	)
+}
+
```

Comment:
This needs more docs, whenever you generate code please add a comment with how the generated code looks like.